framework/src/reporters/ReporterData.C

```diff
@@ -52,6 +52,53 @@ ReporterData::getPostprocessorNames() const
   return output;
 }
 
+DenseVector<Real>
+ReporterData::getAllPostprocessorValues() const
+{
+  DenseVector<Real> all_values;
+
+  std::vector<Real> & output = all_values.get_values();
+
+  for (const auto & name_context_pair : _context_ptrs)
+  {
+    if (name_context_pair.first.isPostprocessor())
+      output.push_back(
+          getReporterValue<PostprocessorValue>(name_context_pair.first.getCombinedName()));
+
+    if (name_context_pair.first.isVectorPostprocessor())
+    {
+      auto & vec =
+          getReporterValue<VectorPostprocessorValue>(name_context_pair.first.getCombinedName());
+      for (const auto & v : vec)
+        output.push_back(v);
+    }
+  }
+
+  return all_values;
+}
+
+std::vector<std::string>
+ReporterData::getAllPostprocessorFullNames() const
+{
+  std::vector<std::string> output;
+
+  for (const auto & name_context_pair : _context_ptrs)
+  {
+    if (name_context_pair.first.isPostprocessor())
+      output.push_back(name_context_pair.first.getCombinedName());
+
+    if (name_context_pair.first.isVectorPostprocessor())
+    {
+      auto pname = name_context_pair.first.getCombinedName();
+      auto & vec = getReporterValue<VectorPostprocessorValue>(pname);
+      for (unsigned int i = 0; i < vec.size(); ++i)
+        output.push_back(pname + "/" + std::to_string(i));
+    }
```

Comment:
```suggestion
    const ReporterName & rname = name_context_pair.first;
    
    if (hasReporterValue<Real>(rname))
      output.push_back(rname);

    if (hasReporterValue<std::vector<Real>>(rname))
    {
      const auto & vec = getReporterValue<std::vector<Real>>(rname);
      for (unsigned int i = 0; i < vec.size(); ++i)
        output.push_back(std::to_string(rname) + "_" + std::to_string(i));
    }
```

Reply:
I see what you want. I will need to change both functions and possibly rename the function a bit to reflect this change.

Reply:
`/` makes the print nicer than `_` because it clearly separate the vector name and the index, otherwise the index looks like part of the name.